.github/workflows/pr-ci.yml

```diff
@@ -73,9 +73,47 @@ jobs:
     with:
       checkout-repo: ${{ needs.checkout-branch.outputs.PR_repo }}
       checkout-ref: ${{ needs.checkout-branch.outputs.PR_head_ref }}
+
+  start-comment:
+    runs-on: ubuntu-latest
+    needs: [check-before-build]
+    if: ${{ needs.check-before-build.output.result == 'success' }}
+    steps:
+      - name: Capture Date and Time
+        id: date-time
+        run: echo "::set-output name=date-time::$(date +'%Y-%m-%d %H:%M:%S')"
+      - name: Start Comment
+        uses: thollander/actions-comment-pull-request@v2
+        with:
+          message: |
+            Workflow started at ${{steps.date-time.outputs.date-time}}.
+            Workflow run [Link](https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}).
+
       
   build-and-test:
-    needs: [code-analysis, checkout-branch]
+    needs: [checkout-branch]
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: build-test
+        uses: ./.github/workflows/ci-build-image.yml
+        with:
+         build-arch: ${{ matrix.arch }}
+         checkout-repo: ${{ needs.checkout-branch.outputs.PR_repo }}
+         checkout-ref: ${{ needs.checkout-branch.outputs.PR_head_ref  }}
+         skip-itests: ${{ matrix.arch != 'amd64' }}
+         run-tests-only: true
+      - name: Set BUILD_TEST_EXECUTED flag
+        id: set-build-test-executed
+        run: echo "BUILD_TEST_EXECUTED=true" >> $GITHUB_ENV
+
+  retest-integration:
+    needs: [code-analysis, checkout-branch, build-and-test]
+    if: startsWith(github.event.comment.body, '/retest') && env.BUILD_TEST_EXECUTED == 'true'
```

Comment:
I don't know if this `env.BUILD_TEST_EXECUTED` is going to persist between completely different runs - when the workflow is re-triggered by a new `/retest` comment, I would expect that the `$GITHUB_ENV` from the last workflow run is not still here for the new run to read. Maybe instead, this needs to check that it is able to pull a `ghcr.io` test image with the expected image tag?